src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -63,9 +63,60 @@ def float_nan_inf_to_sql(value: float) -> str:
     return f"{cast_value} :: FLOAT"
 
 
-def to_sql(value: Any, datatype: DataType, from_values_statement: bool = False) -> str:
+def to_sql(
+    value: Any,
+    datatype: DataType,
```

Comment:
have you considered `numeric_to_sql_without_cast`

Reply:
also `to_sql_try_avoid_cast`

Reply:
I think this function does not apply here since it has a fallback mechanism in it, I will also need to change it if we want to use it, so maybe just put the logic in `to_sql` is more straight forward